### PR TITLE
Switch Flag "None" macro

### PR DESCRIPTION
**Pull request by engineer124**

### CODE REVIEW

**Review by hensldm — COMMENTED** [review]
Could also do None macros for the individual actors as well? For example quite a few compare their switchflag to `0x7F` to check if it is none.

**Review by Dragorn421 — COMMENTED** [review]
Is there a discussion leading to this pr somewhere?

I'm not convinced about this porting to oot, where setting "no switch flag" in actor params is more often 0x1F or various specific mechanisms

**Comment by hensldm** [comment]
> Is there a discussion leading to this pr somewhere?
> 
> I'm not convinced about this porting to oot, where setting "no switch flag" in actor params is more often 0x1F or various specific mechanisms

That is partly what my comment above was about. A lot of actors don't use a full length -1, instead use -1 of various widths (e.g. `0x1F` is a 5 bits -1), so it is nice to have these individual actor NONE's. If you want a more concrete example that has already been merge look at PATH_NONE. There are currently 2 overarching NONE's, and than each actor that needs one gets a specific PATH_NONE.

**Comment by engineer124** [comment]
The None macros for the individual actors have been added, you can see how it looks overall for the discussion now

File: src/overlays/actors/ovl_Elf_Msg2/z_elf_msg2.c [hunk 1]
```diff
@@ -135,7 +135,7 @@ void func_8096EF98(ElfMsg2* this, PlayState* play) {
 
 void func_8096EFD0(ElfMsg2* this, PlayState* play) {
     if ((this->actor.home.rot.y < 0) && (this->actor.home.rot.y >= -0x80) &&
-        (Flags_GetSwitch(play, -this->actor.home.rot.y - 1))) {
+        Flags_GetSwitch(play, -this->actor.home.rot.y - 1)) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_Elf_Msg2/z_elf_msg2.c:138]: I wonder if `SWITCH_FLAG_NONE` could be used here in any way.

File: src/overlays/actors/ovl_Elf_Msg2/z_elf_msg2.c [hunk 2]
```diff
@@ -41,7 +41,7 @@ void ElfMsg2_SetupAction(ElfMsg2* this, ElfMsg2ActionFunc actionFunc) {
 
 s32 func_8096EC4C(ElfMsg2* this, PlayState* play) {
     if ((this->actor.home.rot.y > 0) && (this->actor.home.rot.y < 0x81) &&
-        (Flags_GetSwitch(play, this->actor.home.rot.y - 1))) {
+        Flags_GetSwitch(play, this->actor.home.rot.y - 1)) {
```
Comment (mzxrules) [inline src/overlays/actors/ovl_Elf_Msg2/z_elf_msg2.c:44]: that line above should be <= 0x80

File: src/overlays/actors/ovl_Elf_Msg2/z_elf_msg2.c (hunk 1 above)
Comment (mzxrules) [reply src/overlays/actors/ovl_Elf_Msg2/z_elf_msg2.c:138]: `this->actor.home.rot.y` cannot be 0 per the condition, and a value of -0x80 "sets" SWITCH_FLAG_NONe I believe.

**Review by tom-overton — APPROVED** [review]

**Pull request merged**